### ISSUE TITLE
weather modules is broken

### ISSUE THREAD

**Issue by RustyBower** [sopel-irc/sopel#1435]
As of January 3, 2019, Yahoo has deprecated the use of its public weather API (https://developer.yahoo.com/weather/). I haven't found a good public weather API, but I found a suitable alternative in OpenWeatherMap. Unfortunately, this requires an API key, so probably isn't suitable for a default sopel module.

I've created a sopel module for this at https://github.com/RustyBower/sopel-weather

**Comment by dgw** [comment]
I did some research on this a while back. Basically, free weather APIs have become extinct due to massive amounts of consolidation in the industry.

OpenWeatherMap is probably the best alternative, but you're correct that it won't go into Sopel core because of the API key requirement. Your standalone module will be mentioned in the changelog for 6.6.0. Feel free to also create a PR that removes `weather.py` from this repo.

You might also consider publishing your weather module to PyPI, to make it easier to install. If you ever get tired of maintaining it, there will always be a standing offer for the sopel-irc org to adopt it, as was done with e.g. sopel-github. :smile_cat:

**Comment by dgw** [comment]
Oh, also, the core `weather` module gained a forecast command that was never released before the API deprecation (#1187). Maybe you can adapt the code for yours, at some point.

I'm working on drafting the changelog for 6.6.0 right now (which is what reminded me of the now-useless new forecast command), and I'd love to just give a PyPI package name right alongside the removal note. Feel free to ping me in Sopel's IRC channel or something.

**Comment by RustyBower** [comment]
Yep. I saw the changes in the weather module while I was porting everything over to the new API. I'll implement the forecast stuff this weekend and get everything buttoned up into PyPI

**Comment by dgw** [comment]
@RustyBower Just wanted to make sure you're aware of another OWM module recently created, https://github.com/davidpirogov/sopel-openweathermap

Maybe you can work together, I don't know. I just don't think it would be good to have two replacements.

**Comment by RustyBower** [comment]
I'll reach out. The main issues I see are the lack of forecast stuff, and the fact that it's not in module form right now.

**Comment by RustyBower** [comment]
Either way, weather module is now up on PyPI (https://pypi.org/project/sopel-modules.weather/)

**Comment by RustyBower** [comment]
We should be all set now! PyPI description was cleaned up (thanks for the link on that, I never knew how to fix it before), and we have .forecast functionality!

**Comment by dgw** [comment]
There's no development reason to leave this issue open, but I might as well pin it until 6.6.0 is released for visibility. Users looking for a working weather module can find it more easily this way.
